diff --git a/python-package/setup.py b/python-package/setup.py
index 8a1b1b70..38d35bfe 100644
--- a/python-package/setup.py
+++ b/python-package/setup.py
@@ -345,7 +345,6 @@ if __name__ == '__main__':
           cmdclass={
               'build_ext': BuildExt,     # type: ignore
               'sdist': Sdist,            # type: ignore
-              'install_lib': InstallLib,  # type: ignore
               'install': Install          # type: ignore
           },
           extras_require={
diff --git a/python-package/xgboost/libpath.py b/python-package/xgboost/libpath.py
index 1ae77556..d5fa594f 100644
--- a/python-package/xgboost/libpath.py
+++ b/python-package/xgboost/libpath.py
@@ -21,6 +21,7 @@ def find_lib_path() -> List[str]:
     """
     curr_path = os.path.dirname(os.path.abspath(os.path.expanduser(__file__)))
     dll_path = [
+		'/usr/lib',
         # normal, after installation `lib` is copied into Python package tree.
         os.path.join(curr_path, 'lib'),
         # editable installation, no copying is performed.
