diff --git a/cmake/Modules/FindHDFS.cmake b/cmake/Modules/FindHDFS.cmake
index dcc547f..5f65abc 100644
--- a/cmake/Modules/FindHDFS.cmake
+++ b/cmake/Modules/FindHDFS.cmake
@@ -44,10 +44,6 @@ find_library(HDFS_LIB NAMES hdfs PATHS
 if (HDFS_LIB)
   set(HDFS_FOUND TRUE)
   set(HDFS_LIBRARIES ${HDFS_LIB})
-  set(HDFS_STATIC_LIB ${HDFS_LIB_PATHS}/${CMAKE_STATIC_LIBRARY_PREFIX}hdfs${CMAKE_STATIC_LIBRARY_SUFFIX})
-
-  add_library(hdfs_static STATIC IMPORTED)
-  set_target_properties(hdfs_static PROPERTIES IMPORTED_LOCATION ${HDFS_STATIC_LIB})
 
 else ()
   set(HDFS_FOUND FALSE)
@@ -58,7 +54,6 @@ if (HDFS_FOUND)
     message(STATUS "${Hadoop_VERSION}")
     message(STATUS "HDFS_INCLUDE_DIR: ${HDFS_INCLUDE_DIR}")
     message(STATUS "HDFS_LIBRARIES: ${HDFS_LIBRARIES}")
-    message(STATUS "hdfs_static: ${HDFS_STATIC_LIB}")
   endif ()
 else ()
   message(FATAL_ERROR "HDFS includes and libraries NOT found."
@@ -68,5 +63,4 @@ endif ()
 mark_as_advanced(
   HDFS_LIBRARIES
   HDFS_INCLUDE_DIR
-  hdfs_static
 )
