diff --git a/src/xfce-desktop.c b/src/xfce-desktop.c
index 40d74037..9f3acf52 100644
--- a/src/xfce-desktop.c
+++ b/src/xfce-desktop.c
@@ -449,12 +449,14 @@ backdrop_changed_cb(XfceBackdrop *backdrop, gpointer user_data)
 {
     XfceDesktop *desktop = XFCE_DESKTOP(user_data);
     cairo_surface_t *surface = desktop->priv->bg_surface;
+    cairo_surface_t *pix_surface;
     GdkScreen *gscreen = desktop->priv->gscreen;
     GdkDisplay *display;
     gchar *new_filename = NULL;
     GdkRectangle rect;
     cairo_region_t *clip_region = NULL;
     gint i, monitor = -1, current_workspace;
+    gint scale;
 #ifdef G_ENABLE_DEBUG
     gchar *monitor_name = NULL;
 #endif
@@ -536,6 +538,14 @@ backdrop_changed_cb(XfceBackdrop *backdrop, gpointer user_data)
                  rect.x, rect.y, rect.width, rect.height);
     }
 
+    scale = gdk_window_get_scale_factor(gdk_screen_get_root_window(gscreen));
+    scale = scale ? scale : 1;
+
+    rect.x *= 1;
+    rect.y *= 1;
+    rect.width *= scale;
+    rect.height *= scale;
+
     xfce_backdrop_set_size(backdrop, rect.width, rect.height);
 
     if(monitor > 0
@@ -605,7 +615,9 @@ backdrop_changed_cb(XfceBackdrop *backdrop, gpointer user_data)
         }
 
         cr = cairo_create(surface);
-        gdk_cairo_set_source_pixbuf(cr, pix, rect.x, rect.y);
+        pix_surface = gdk_cairo_surface_create_from_pixbuf(pix, scale, NULL);
+        cairo_set_source_surface(cr, pix_surface, rect.x, rect.y);
+        cairo_surface_destroy(pix_surface);
 
         /* clip the area so we don't draw over a previous wallpaper */
         if(clip_region != NULL) {
